tests/unit/models/test_projects.py

```diff
@@ -710,6 +710,66 @@ def test_platforms_not_allowed_core22(self, project_yaml_data):
             "Use 'architectures' keyword instead." in str(raised.value)
         )
 
+    @pytest.mark.parametrize(
+        ("architectures", "expected"),
+        [
+            ([], {}),
+            (
+                ["amd64"],
+                {
+                    "amd64": Platform(
+                        build_for=[const.SnapArch("amd64")],
+                        build_on=[const.SnapArch("amd64")],
+                    )
+                },
+            ),
+            (
+                [Architecture(build_on="amd64", build_for="riscv64")],
+                {
+                    "riscv64": Platform(
+                        build_for=[const.SnapArch("riscv64")],
+                        build_on=[const.SnapArch("amd64")],
+                    )
+                },
+            ),
+            (
+                [
+                    Architecture.unmarshal(
+                        {"build_on": ["amd64"], "build_for": ["riscv64"]}
+                    )
+                ],
+                {
+                    "riscv64": Platform(
+                        build_for=[const.SnapArch("riscv64")],
+                        build_on=[const.SnapArch("amd64")],
+                    )
+                },
+            ),
+            (
+                [
+                    Architecture.unmarshal(
+                        {"build_on": ["amd64", "arm64"], "build_for": ["riscv64"]}
+                    ),
+                    Architecture.unmarshal(
+                        {"build_on": ["amd64", "arm64"], "build_for": ["arm64"]}
+                    ),
+                ],
+                {
+                    "riscv64": Platform(
+                        build_for=[const.SnapArch("riscv64")],
+                        build_on=[const.SnapArch("amd64"), const.SnapArch("arm64")],
+                    ),
+                    "arm64": Platform(
+                        build_for=[const.SnapArch("arm64")],
+                        build_on=[const.SnapArch("amd64"), const.SnapArch("arm64")],
+                    ),
+                },
+            ),
+        ],
+    )
+    def test_from_architectures(self, architectures, expected):
```

Comment:
This test is fine, but with all the validators and such I think we might want to have a (parametrized) test specifically for `get_build_plan()` values for `core22` projects. What do you think?

Reply:
You're right, and implementing this test caught a bug!